### ISSUE TITLE
`reset()` não limpa o replay buffer nem o preprocessador

### ISSUE THREAD

**Issue by luizth** [luizth/ppgca-dl-dqn#9]
## Onde

`src/agent.py:87-95`

```python
def reset(self):
    super().reset()
    self.Q = self._Q_initial.copy()
    self.Q_target = self._Q_initial.copy()
    self.optimizer = optim.SGD(self.Q.parameters(), lr=self.learning_rate)
    self._current_state, _ = self._env.reset()
```

## Causa

`reset()` restaura rede, rede alvo, otimizador, epsilon e o estado do env, mas **não** limpa:

- `self.eb` — o replay buffer mantém transições geradas pela política antiga, e os `state`/`next_state` guardados são features codificadas por uma rede que não existe mais;
- `self.preprocessor` — os buffers de frames continuam cheios (relacionado a #7).

Ou seja, `reset()` não devolve o agente ao estado inicial. Hoje `main.py:154` chama `reset()` logo após a construção, quando o buffer já está vazio, então o bug está latente — mas qualquer re-run ou seed múltiplo no mesmo processo pega o problema.

## Correção

```python
self.eb = ExperienceBuffer(self.eb.max_lenght)
if self.preprocessor is not None:
    self.preprocessor.reset()
```
